src/api/java/CMakeLists.txt

```diff
@@ -20,9 +29,35 @@ add_custom_target(
     "${PYTHON_EXECUTABLE}"
     "${CMAKE_CURRENT_LIST_DIR}/genkinds.py"
     --kinds-header "${PROJECT_SOURCE_DIR}/src/api/cvc4cppkind.h"
-    --kinds-file-prefix "${CMAKE_CURRENT_BINARY_DIR}/cvc/Kind"
+    --kinds-file-prefix "${CMAKE_CURRENT_BINARY_DIR}/src/main/java/cvc/Kind"
   DEPENDS
     genkinds.py
   COMMENT
     "Generate Kind.java"
 )
+
+
+find_package(Java)
+include(UseJava)
+find_package(JNI REQUIRED)
+
+# specify the output destination of cvcJavaApi library
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/src/main/resources")
+
+add_library(cvcJavaApi SHARED jni/cvc_Solver.cpp)
+
+target_include_directories(cvcJavaApi PUBLIC ${JNI_INCLUDE_DIRS})
+target_include_directories(cvcJavaApi PUBLIC "${CMAKE_INSTALL_PREFIX}/include")
+target_link_libraries(cvcJavaApi ${JNI_LIBRARIES})
+target_link_libraries(cvcJavaApi cvc4)
+
+# run gradle
+add_custom_target(
+  gradleBuild
+  POST_BUILD
+  COMMAND
+    "cd ${CMAKE_CURRENT_BINARY_DIR}"
```

Comment:
There should be no need for the `cd`. You should be already in `CMAKE_CURRENT_BINARY_DIR`.

Reply:
Yes. This custom command does not execute for some reason. I added this line to see if it fixes the issue, but it did not. I removed it now.

Reply:
Fixed